pkg/controllers/tlspolicy/tlspolicy_controller.go

```diff
@@ -157,17 +162,33 @@ func (r *TLSPolicyReconciler) reconcileResources(ctx context.Context, tlsPolicy
 		return err
 	}
 
-	if err := r.reconcileCertificates(ctx, tlsPolicy, gatewayDiffObj); err != nil {
-		return err
+	if err = r.reconcileCertificates(ctx, tlsPolicy, gatewayDiffObj); err != nil {
+		gatewayCondition = conditions.BuildPolicyAffectedCondition(TLSPolicyAffected, tlsPolicy, targetNetworkObject, conditions.PolicyReasonInvalid, err)
+		updateErr := r.updateGatewayCondition(ctx, gatewayCondition, gatewayDiffObj)
+		return errors.Join(fmt.Errorf("reconcile Certificates error %w", err), updateErr)
 	}
 
 	// set direct back ref - i.e. claim the target network object as taken asap
-	if err := r.ReconcileTargetBackReference(ctx, client.ObjectKeyFromObject(tlsPolicy), targetNetworkObject, TLSPolicyBackRefAnnotation); err != nil {
-		return err
+	if err = r.ReconcileTargetBackReference(ctx, client.ObjectKeyFromObject(tlsPolicy), targetNetworkObject, TLSPolicyBackRefAnnotation); err != nil {
+		gatewayCondition = conditions.BuildPolicyAffectedCondition(TLSPolicyAffected, tlsPolicy, targetNetworkObject, conditions.PolicyReasonConflicted, err)
+		updateErr := r.updateGatewayCondition(ctx, gatewayCondition, gatewayDiffObj)
+		return errors.Join(fmt.Errorf("reconcile TargetBackReference error %w", err), updateErr)
+	}
+
+	// set annotation of policies affecting the gateway
+	if err = r.ReconcileGatewayPolicyReferences(ctx, tlsPolicy, gatewayDiffObj); err != nil {
+		gatewayCondition = conditions.BuildPolicyAffectedCondition(TLSPolicyAffected, tlsPolicy, targetNetworkObject, conditions.PolicyReasonUnknown, err)
+		updateErr := r.updateGatewayCondition(ctx, gatewayCondition, gatewayDiffObj)
+		return errors.Join(fmt.Errorf("ReconcileGatewayPolicyReferences error %w", err), updateErr)
 	}
 
-	// set annotation of policies affecting the gateway - should be the last step, only when all the reconciliation steps succeed
-	return r.ReconcileGatewayPolicyReferences(ctx, tlsPolicy, gatewayDiffObj)
+	// set gateway policy affected condition status - should be the last step, only when all the reconciliation steps succeed
+	updateErr := r.updateGatewayCondition(ctx, gatewayCondition, gatewayDiffObj)
+	if updateErr != nil {
+		return fmt.Errorf("failed to update gateway conditions %w ", updateErr)
+	}
+
+	return nil
 }
```

Comment:
Note, this is pretty much copied from the DNSPolicy for consistency. Any suggestions that might be made will need to be reflected there also if applicable.